pkgs/development/libraries/gdal/default.nix

```diff
@@ -3,7 +3,8 @@
 , libpng
 , netcdf, hdf5 , curl
 , netcdfSupport ? true
- }:
+, pythonSupport ? !stdenv.isDarwin
+}:
```

Comment:
Could you change this to `true` and pass `pythonSupport = !stdenv.isDarwin;` in `all-packages.nix`. In `python-packages.nix` we override `gdal` to use the version of the Python interpreter that specific set is build for. Those builds should fail and not silently build without Python.